inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp

```diff
@@ -36,26 +35,36 @@ void MKLDNNSplitNode::getSupportedDescriptors() {
 }
 
 void MKLDNNSplitNode::initSupportedPrimitiveDescriptors() {
+    constexpr size_t expectedPdVectorSize = 5ul;
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
-    InferenceEngine::Precision precision = getCnnLayer()->insData[0].lock()->getPrecision();
-    if (precision != InferenceEngine::Precision::FP32)
-        precision = InferenceEngine::Precision::FP32;
-    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
-    precision = getCnnLayer()->outData[0]->getPrecision();
-    if (precision != InferenceEngine::Precision::FP32)
-        precision = InferenceEngine::Precision::FP32;
-    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+    if (getCnnLayer()->insData.empty()) {
+        THROW_IE_EXCEPTION << "Split node '" << getName() << "' has an empty input in the CNN layer";
+    }
+
+    auto inpData = getCnnLayer()->insData[0].lock();
+    if (!inpData) {
+        THROW_IE_EXCEPTION << "Split node '" << getName() << "' input data is empty";
+    }
+
+    supportedPrimitiveDescriptors.reserve(expectedPdVectorSize);
 
+    InferenceEngine::Precision inpPrecision = inpData->getPrecision();
+    auto outPrecision = inpPrecision; // the split layer doesn't convert precisions
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inpPrecision);
+    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(outPrecision);
+
+    //Set plain format
     auto srcDims = getParentEdgeAt(0)->getDims();
+    auto memoryFormat = MKLDNNMemory::GetPlainFormat(srcDims);
 
     InferenceEngine::LayerConfig config;
     config.dynBatchSupport = true;
```

Comment:
What about case when split axis == 0? I think should disable dynamic batch feature in this case.

Reply:
Additional check have been added in ee8234d71967059ae0d72418abd7796120119305.